carts/views.py

```diff
@@ -0,0 +1,234 @@
+"""
+Views for carts app
+"""
+# pylint: disable= no-self-use, no-member
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .constants import OPEN, SUBMITTED
+from .serializers import Cart, CartItem, CartItemSerializer, CartSerializer
+
+
+class CartsAPIView(APIView):
+    """
+    Carts APIView allows a user to add to cart, update a cart item, remove a cart item,
+    view all or one cart item and clear all the cart items
+    """
+
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        """
+        Adds a cart_item to cart
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): Value containing information about operation status
+        """
+        user = request.user
+        try:
+            cart = get_object_or_404(Cart, user=user, status=OPEN)
+        except Http404:
+            cart_serializer = CartSerializer(data={"user": user.id})
+            if cart_serializer.is_valid():
```

Comment:
can we use get_or_create() instead of 404 exception catch? Or maybe try to get the cart and if it doesn't exist we create one. It is not a good idea to raise and then catch 404 yourself.

Reply:
Done